jgrapht-core/src/main/java/org/jgrapht/alg/planar/ElementList.java

```diff
@@ -0,0 +1,438 @@
+/*
+ * (C) Copyright 2018-2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.planar;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * This is the implementation of the doubly linked list data structure, which gives access to the internal
+ * list nodes the data is stored in. The primary goal of this is to be able to delete an element from this
+ * list in $\mathcal{O}(1)$ via the list node it is stored in.
+ * <p>
+ * Currently, this data structure doesn't have a fail-fast behavior, meaning that the user have to ensure,
+ * that the list isn't modified during the iteration over its elements. Otherwise, the result of such
+ * iteration is undefined.
+ *
+ * @param <E> the list element type
+ * @author Timofey Chudakov
+ */
+public class ElementList<E> implements Iterable<E> {
```

Comment:
If we are going to keep this, I suggest to add a `modCount` just like `LinkedList` and throw a `ConcurrentModificationException` when detected.